### PR TITLE
Wrap aecore EUnit tests as CT suite

### PR DESCRIPTION
**Pull request by uwiger**
A parse transform locates EUnit tests and fills in a Common Test skeleton suite.

This seems to fix some of the cleanup issues with EUnit, as well as give better overview of the outcome.

TODO: add some lines in `.travis.yml` to fetch CT info if the test fails. [This](https://github.com/GENIVI/rvi_core/blob/develop/.travis.yml) might serve as inspiration.

### CODE REVIEW

**Comment by uwiger** [comment]
Note that `aec_eunit_SUITE.erl` and `ct_eunit_xform.erl` in `apps/aecore/test/` are generic and should work if they’re dropped into the other apps.

The result of the transform can be seen in:
`_build/test/lib/aecore/test/aec_eunit_SUITE.xfm` (which is debug output generated by `parse_trans`).

File: apps/aecore/test/ct_eunit_xform.erl [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            parse_trans:do_insert_forms(
+              below,
+              add_functions(AllMods),
+              NewForms,
+              Ctxt),
+            Ctxt),
+    parse_trans:optionally_pretty_print(Res, Opts, Ctxt),
+    Res.
+
+xform({function,_,groups,0,_} = Form, AllMods) ->
```
Comment (happi) [inline apps/aecore/test/ct_eunit_xform.erl]: Form is unused.

**Comment by lucafavatella** [comment]
> This seems to fix some of the cleanup issues with EUnit ...

@uwiger Any specific cleanup issues with EUnit in current master? Locally I ran tests on master and I did not experience timeouts.

Locally running `epmd -daemon && make test` on this PR (f8398f2) passed for me. Though I see [CI](https://travis-ci.org/aeternity/epoch/jobs/295687632#L1331) failed: not sure why test attempts "upgrade" on CI and not on my local machine (need to investigate):
```
%%% aec_eunit_SUITE ==> aec_tx_pool_tests: OK

=ERROR REPORT==== 1-Nov-2017::10:59:37 ===

Error in process <0.4967.0> on node 'epoch_ct@travis-job-aeternity-epoch-295687632' with exit value:

{{badmatch,{error,{upgrade,"Upgrade not supported by this NIF library."}}},

 [{aec_pow_cuckoo,init,0,

                  [{file,"/home/travis/build/aeternity/epoch/_build/test/lib/aecore/src/aec_pow_cuckoo.erl"},

                   {line,37}]},

  {code_server,'-handle_on_load/5-fun-0-',1,

               [{file,"code_server.erl"},{line,1333}]}]}

=WARNING REPORT==== 1-Nov-2017::10:59:37 ===

The on_load function for module aec_pow_cuckoo returned:

{{badmatch,{error,{upgrade,"Upgrade not supported by this NIF library."}}},

 [{aec_pow_cuckoo,init,0,[{file,...},{...}]},

  {code_server,'-handle_on_load/5-fun-0-',1,[{...}|...]}]}

%%% aecore_sync_SUITE ==> init_per_suite: FAILED

%%% aecore_sync_SUITE ==> {error_loading_module,aec_pow_cuckoo,on_load_failure}

%%% aecore_sync_SUITE ==> start_first_node (group two_nodes): SKIPPED

%%% aecore_sync_SUITE ==> {tc_auto_skip,

    {failed,

        {aecore_sync_SUITE,init_per_suite,

            {'EXIT',{error_loading_module,aec_pow_cuckoo,on_load_failure}}}}}
```

**Comment by uwiger** [comment]
@lucafavatella, yes these were the errors I had seen before. I was a bit surprised that they didn't hit when I tested locally today.

**Comment by lucafavatella** [comment]
> Though I see CI failed: not sure why test attempts "upgrade" on CI and not on my local machine (need to investigate)

I [opened PR](https://github.com/aeternity/epoch/pull/313) against this branch for fixing this.

**Comment by lucafavatella** [comment]
@uwiger Would you please consider merging PR #313 into this PR (i.e. clicking "merge PR" in PR 313)? It shall help move this PR 312 forward. Cc @gorillainduction

**Pull request merged**